### PR TITLE
paperless-ng: disable broken "slow_write_pdf" tests

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change

The [test_slow_write_pdf test](https://github.com/jonaswinkler/paperless-ng/blob/ng-1.5.0/src/documents/tests/test_management_consumer.py#L135) fails to compare the "slowly" generated file in https://github.com/jonaswinkler/paperless-ng/blob/ng-1.5.0/src/documents/tests/test_management_consumer.py#L79

Depends on #158111
See also #136626

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Flakebi — APPROVED** [review]
Nice!
Still can’t reproduce the failure, but I checked that paperless-ng builds locally and nixos tests are passing, so thanks a lot.

File: pkgs/applications/office/paperless-ng/default.nix [hunk 1]
```diff
@@ -172,7 +172,8 @@ py.pkgs.pythonPackages.buildPythonApplication rec {
   # - XDG_DATA_DIRS with test-specific fonts
```
Comment (erikarvstedt) [inline pkgs/applications/office/paperless-ng/default.nix:172]:
```suggestion
  # - XDG_DATA_DIRS with test-specific fonts
  #
  # Skip test `slow_write_pdf` which fails on some systems:
  # https://github.com/NixOS/nixpkgs/issues/136626
```

**Review by erikarvstedt — APPROVED** [review]
* [x] Build succeeds on `x86_64-linux`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
We should use `pytestCheckHook`.

**Comment by jojosch** [comment]
Now using `pytestCheckHook`. 

Also applied the patch from @gador instead of disabling the `slow_write_pdf` test. Thank you.

File: pkgs/applications/office/paperless-ng/default.nix [hunk 2]
```diff
@@ -54,6 +55,15 @@ py.pkgs.pythonPackages.buildPythonApplication rec {
     sha256 = "oVSq0AWksuWC81MF5xiZ6ZbdKKtqqphmL+xIzJLaDMw=";
   };
 
+  patches = [
+    # Fix the `slow_write_pdf` test:
+    # https://github.com/NixOS/nixpkgs/issues/136626
+    (fetchpatch {
+      url = "https://github.com/paperless-ngx/paperless-ngx/pull/66/commits/4fbabe43ea12811864e9676b04d82a82b38e799d.patch";
```
Comment (dotlambda) [inline pkgs/applications/office/paperless-ng/default.nix:62]:
```suggestion
      url = "https://github.com/paperless-ngx/paperless-ngx/commit/4fbabe43ea12811864e9676b04d82a82b38e799d.patch";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
[Fixup](https://github.com/erikarvstedt/nixpkgs/commits/paperless-ng-disable-broken-tests-ea).
(Fetch with `git fetch https://github.com/erikarvstedt/nixpkgs paperless-ng-disable-broken-tests-ea`.)
This makes it clear that the check phase is run after the install phase and that the PWD change (`cd src`) in `preCheck` has no effect on the install phase.

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
Please squash obsolete commit `paperless-ng: disable broken "slow_write_pdf" tests`.

**Comment by jojosch** [comment]
Redone the commits, moved `checkPhase` and added a comment to the check phase regarding the `cd src`.

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
@jojosch, thanks!
The comment `working directory changed to ./src for the duration of the checkPhase` is incorrect because PWD is not reset after `pytestCheckPhase`. 
I'd change the comment [like this](https://github.com/erikarvstedt/nixpkgs/commits/paperless-ng-disable-broken-tests-ea-2).

**Comment by jojosch** [comment]
> I'd change the comment [like this](https://github.com/erikarvstedt/nixpkgs/commits/paperless-ng-disable-broken-tests-ea-2).

done

**Review by erikarvstedt — APPROVED** [review]

**Review by erikarvstedt — CHANGES_REQUESTED** [review]
Sorry, one last change:
It's simpler and more robust to just add a `postCheck` instead of a comment. [Fixup](https://github.com/erikarvstedt/nixpkgs/commits/paperless-ng-disable-broken-tests-ea-3).

**Review by erikarvstedt — APPROVED** [review]

**Comment by mweinelt** [comment]
- `doCheck` defaults to true for python packaging
- order the `installPhase` before anything check-related
- instead of `cd src` use `pytestFlagsArray` to pass `src` as the path to find the tests below
- Drop `pytest` because it is included in `pytestCheckHook
- Drop `pytest-cov` because only upstream cares about test coverage, for us it is noise at best

```patch
diff --git a/pkgs/applications/office/paperless-ng/default.nix b/pkgs/applications/office/paperless-ng/default.nix
index 88f546d9aef..15b129ae2a5 100644
--- a/pkgs/applications/office/paperless-ng/default.nix
+++ b/pkgs/applications/office/paperless-ng/default.nix
@@ -165,10 +165,16 @@ py.pkgs.pythonPackages.buildPythonApplication rec {
     zope_interface
   ];
 
-  doCheck = true;
+  installPhase = ''
+    mkdir -p $out/lib
+    cp -r . $out/lib/paperless-ng
+    chmod +x $out/lib/paperless-ng/src/manage.py
+    makeWrapper $out/lib/paperless-ng/src/manage.py $out/bin/paperless-ng \
+      --prefix PYTHONPATH : "$PYTHONPATH" \
+      --prefix PATH : "${path}"
+  '';
+
   checkInputs = with py.pkgs.pythonPackages; [
-    pytest
-    pytest-cov
     pytest-django
     pytest-env
     pytest-sugar
@@ -177,14 +183,7 @@ py.pkgs.pythonPackages.buildPythonApplication rec {
     pytestCheckHook
   ];
 
-  installPhase = ''
-    mkdir -p $out/lib
-    cp -r . $out/lib/paperless-ng
-    chmod +x $out/lib/paperless-ng/src/manage.py
-    makeWrapper $out/lib/paperless-ng/src/manage.py $out/bin/paperless-ng \
-      --prefix PYTHONPATH : "$PYTHONPATH" \
-      --prefix PATH : "${path}"
-  '';
+  pytestFlagsArray = [ "src "];
 
   # The tests require:
   # - PATH with runtime binaries
@@ -195,8 +194,8 @@ py.pkgs.pythonPackages.buildPythonApplication rec {
     export HOME=$(mktemp -d)
     export XDG_DATA_DIRS="${liberation_ttf}/share:$XDG_DATA_DIRS"
 
-    # This doesn't affect the `installPhase` because `pytestCheckPhase` is run afterwards
-    cd src
+    substituteInPlace src/setup.cfg \
+      --replace "--cov --cov-report=html" ""
   '';
 
   passthru = {
```

**Comment by erikarvstedt** [comment]
@mweinelt, great, thanks! I've resolved the merge conflict and the minor whitespace issue (`"src "`) in this [fixup](https://github.com/erikarvstedt/nixpkgs/commits/paperless-ng-disable-broken-tests-ea-4).

**Review by erikarvstedt — APPROVED** [review]

**Pull request merged**